src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -245,51 +246,64 @@ def _get_resource(client, resource_group_name, *subresources):
             raise
 
 
-def _get_vnet_network_profile(cmd, location, resource_group_name, vnet_name, vnet_address_prefix, subnet, subnet_address_prefix):
+def _get_vnet_network_profile(cmd, location, resource_group_name, vnet, vnet_address_prefix, subnet, subnet_address_prefix):
     from azure.cli.core.profiles import ResourceType
     from msrestazure.tools import parse_resource_id, is_valid_resource_id
 
     containerInstanceDelegationServiceName = "Microsoft.ContainerInstance/containerGroups"
     Delegation = cmd.get_models('Delegation', resource_type=ResourceType.MGMT_NETWORK)
     aci_delegation = Delegation(
-        name="Microsoft.ContainerInstance.containerGroups",
-        service_name="Microsoft.ContainerInstance/containerGroups"
+        name=containerInstanceDelegationServiceName,
+        service_name=containerInstanceDelegationServiceName
     )
 
     ncf = cf_network(cmd.cli_ctx)
 
+    vnet_name = vnet
     subnet_name = subnet
     if is_valid_resource_id(subnet):
         parsed_subnet_id = parse_resource_id(subnet)
         subnet_name = parsed_subnet_id['resource_name']
         vnet_name = parsed_subnet_id['name']
+        resource_group_name = parsed_subnet_id['resource_group']
+    elif is_valid_resource_id(vnet):
+        parsed_vnet_id = parse_resource_id(vnet)
+        vnet_name = parsed_vnet_id['resource_name']
+        resource_group_name = parsed_vnet_id['resource_group']
+        ncf.virtual_networks.get(resource_group_name, vnet_name)
```

Comment:
Why are you doing a get without storing it anywhere?  If this is an "existence check" we don't do that in the CLI.  If an ID is valid, we pass it as if it were valid. The service should provide the correct error if it isn't. Existence checking with get is a waste of API quota because if the ID is legitimate then the check wasn't necessary and if it isn't then it would fail regardless.

Reply:
I'll remove